test/e2e/framework/framework.go

```diff
@@ -521,6 +523,17 @@ func (f *Framework) CreateNamespace(baseName string, labels map[string]string) (
 	if createTestingNS == nil {
 		createTestingNS = CreateTestingNS
 	}
+
+	if labels == nil {
+		labels = make(map[string]string)
+	}
```

Comment:
copy the labels arg so we don't modify a map a test might be reusing:

```suggestion
	if labels == nil {
		labels = make(map[string]string)
	} else {
	    labelsCopy = make(map[string]string)
	    for k, v := range labels {
	      labelsCopy[k] = v
	    }
	    labels = labelsCopy
	}
```

Reply:
agreed